airbyte-workers/src/main/java/io/airbyte/workers/helper/FailureHelper.java

```diff
@@ -34,18 +35,46 @@ public static FailureReason genericFailure(final Throwable t, final Long jobId,
         .withMetadata(jobAndAttemptMetadata(jobId, attemptNumber));
   }
 
+  public static FailureReason genericFailure(final AirbyteTraceMessage m, final Long jobId, final Integer attemptNumber) {
+    FailureType failureType;
+    try {
+      final String traceMessageError = m.getError().getFailureType().toString().toUpperCase();
+      failureType = FailureReason.FailureType.valueOf(traceMessageError);
+    } catch (final Exception e) {
+      failureType = FailureType.SYSTEM_ERROR;
+    }
```

Comment:
Based on [this doc](https://docs.google.com/document/d/1ctrj3Yh_GjtQ93aND-WH3ocqGxsmxyC3jfiarrF6NY0/edit#heading=h.uupw5avb79b0), we should default to SYSTEM_ERROR if the failure type is not set on the trace message error at all. However, I think we _should_ set this to `null` / leave it unset on the FailureReason if `valueOf()` fails to translate it from trace to failure reason enum value. 

Also, I think it would be slightly better to use `FailureReason.FailureType.fromValue()` here instead of `valueOf`, since it feels slightly better to match these up based on the values themselves rather than the enum constant names